openeogeotrellis/geopysparkdatacube.py

```diff
@@ -1222,15 +1222,31 @@ def resample_spatial(
             if newLayout == None:
                 return self
 
-            partitioner = max_level.srdd.rdd().partitioner()
-            b = max_level.layer_metadata.bounds
-            rows = b.maxKey.row - b.minKey.row + 1
-            spatialPartitions = (b.maxKey.col - b.minKey.col + 1) * (rows)
+            cellsize_before = self.get_cellsize()
+            if not isinstance(resolution, tuple):
+                resolution = (resolution, resolution)
+
+            estimated_size_in_pixels_tup = self.estimate_layer_size_in_pixels()
 
-            logging.info(f"Reprojecting datacube with partitioner {partitioner} to new layout {newLayout} and {projection}")
-            if(max_level.getNumPartitions() <= spatialPartitions and max_level.layer_type == gps.LayerType.SPACETIME and resolution < 0.5*self.get_cellsize_x()):
-                logging.info(f"Repartitioning datacube with {max_level.getNumPartitions()} partitions to {spatialPartitions*10} before resample_spatial.")
-                max_level = max_level.repartition(spatialPartitions*10)
+            resolution_factor = (
+                cellsize_before[0] / resolution[0],
+                cellsize_before[1] / resolution[1]
+            )
+
+            # If the layer had a smaller tileSize specified, there will probably be enough partitions and this logic won't have effect
+            # If the user has a larger tileSize specified, it will probably be max 1024, then we would make more partitions
+            # Anyway, specifying tileSize is an undocumented feature, so we can assume default tileSize
+            default_tile_size = 256
```

Comment:
@jdries Is the reasoning behind this value ok?

Reply:
256 is not necessarily the default, there's also a feature that adapts it dynamically, even if the user does not set it. 
But I'm not sure why we need to guess it, if we can actually get it from the metadata?
I'm wondering if we should include the number of bands, and then perhaps some guesswork on the temporal dimension, to get a guesstimate of memory per partition?

Reply:
In normal cases, `currentTileLayout.tileCols` could be used, but in `TestMultipleDates.test_resample_spatial` that would be `5`, causing a lot of partitions (slowing the test down x10).

I did not yet find a nice solution that works for normal process graphs and this test. 

For some reason, `layoutCols` is 2 in `test_resample_spatial`, so checking on this value to be 1 to resort to a default won't work

Reply:
I could use tileCols if I just updated the layer in test_resample_spatial to have more realistic proportions. The test suite runs smooth now, as well as the ERA5 resample test.